cluster-autoscaler/core/zalando_simulation_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"flag"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog"
+)
+
+const (
+	labelScalePriority = "zalando.org/scaling-priority"
+)
+
+func TestMain(m *testing.M) {
+	klog.InitFlags(flag.CommandLine)
+	err := flag.Set("v", "3")
+	if err != nil {
+		panic(err)
+	}
+	os.Exit(m.Run())
+}
+
+func TestBrokenScalingTest(t *testing.T) {
+	opts := defaultZalandoAutoscalingOptions()
+	RunSimulation(t, opts, 10*time.Second, func(env *zalandoTestEnv) {
+		env.AddNodeGroup("ng-fallback", 10, resource.MustParse("4"), resource.MustParse("32Gi"), nil)
+		env.AddNodeGroup("ng-1", 10, resource.MustParse("4"), resource.MustParse("32Gi"), map[string]string{labelScalePriority: "100"})
+		env.AddNodeGroup("ng-2", 10, resource.MustParse("4"), resource.MustParse("32Gi"), map[string]string{labelScalePriority: "110"})
+		env.AddNodeGroup("ng-3", 10, resource.MustParse("4"), resource.MustParse("32Gi"), map[string]string{labelScalePriority: "120"})
+
+		env.StepFor(10 * time.Second).ExpectNoCommands()
+
+		env.AddPod(NewTestPod("foo", resource.MustParse("1"), resource.MustParse("24Gi")))
+		env.AddPod(NewTestPod("bar", resource.MustParse("1"), resource.MustParse("24Gi")))
+
+		env.StepUntilCommand(20*time.Hour, zalandoCloudProviderCommand{
+			commandType: zalandoCloudProviderCommandIncreaseSize,
+			nodeGroup:   "ng-fallback",
+			delta:       2,
```

Comment:
I don't understand why this is the expected command. Based on what I can see it should just scale up `ng-3` by 2 and the two pods should be scheduled there.

Reply:
`StepUntilCommand` just runs it until we receive the provided command (with a deadline). Since we don't provide any nodes for the ng-1/2/3 groups, we assume that it will eventually go into the fallback node pool, but it'll take a lot of time.